adagrams/game.py

```diff
@@ -1,11 +1,155 @@
+import random
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+
 def draw_letters():
-    pass
+
+    final_string = []
+    letters_frequency = []
+    letters_list = []
+    
+    for letters in LETTER_POOL:
+        letters_frequency.append(LETTER_POOL[letters])
+    
+
+    for letter in LETTER_POOL:
+        letters_list.append(letter)
```

Comment:
I see that you built two separate lists here so that you can keep track of the number of times that a letter can appear. There are couple of things I would like to point out here.

You are associating a letter in one list with a frequency in another list. Given that we want to associate to pieces of data we usually want to opt for a dictionary here. It makes accessing them easier to get the associated information with a particular key.

Though your code passes the tests and ensuring that no letter appears more than the allowed amount of times, you aren't actually utilizing the weight distribution of the letters. For example, the letter `E` should be 6 times more likely to pulled than the letter `F`. See comment on `Line 46` for a further explanation.

Reply:
So then I should use a list with all the items?